data/src/main/java/com/london/data/datasource/local/recent/search/RecentSearchDataSourceImpl.kt

```
@@ -18,15 +19,15 @@ class RecentSearchDataSourceImpl(
         }
     }
 
-    override suspend fun getAll(): List<RecentSearch> = runCatching {
+    override suspend fun getAll(): List<RecentSearchLocal> = runCatching {
         recentSearchDao.getAll()
     }.getOrDefault(emptyList())
```

why here you are returning an emptyList, i think you should throw a custom exception

---

because i think it does not affect the user but i think i can sent it to firebase what do you think

---

we need to show something to the user so we can do both, we can send the exception to the firebase and domain to display the error screen for example

---

if you return the empty list this is not an indicator that will be a problem so we will not figure out if there was a problem or not

---

yep its background effect